File: services/trader-dashboard/src/types/order.ts
// Order type definitions
import { AssetType } from "./asset-types";

export interface Order {
  id: string;
  symbol: string;
  assetType: AssetType;
  side: "BUY" | "SELL";
  quantity: number;
  price: number;
  orderType: "MARKET" | "LIMIT" | "STOP";
  status: "PENDING" | "FILLED" | "CANCELLED" | "REJECTED";
  timestamp: string;
  filledQuantity?: number;
  averagePrice?: number;
}

export interface PlaceOrderInput {
  symbol: string;
  side: "BUY" | "SELL";
  quantity: number;
  price?: number;
  orderType: "MARKET" | "LIMIT" | "STOP";
}

export interface ModifyOrderInput {
  orderId: string;
  quantity?: number;
  price?: number;
}

export interface OrderResponse {
  success: boolean;
  orderId?: string;
  message?: string;
}

export interface CancelResponse {
  success: boolean;
  message?: string;
}

File: services/dashboard-gateway/src/main/java/com/quantis/dashboard_gateway/model/Order.java
package com.quantis.dashboard_gateway.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Order Model
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    private String orderId;
    private String symbol;
    private String side;
    private BigDecimal quantity;
    private BigDecimal price;
    private String orderType;
    private String status;
    private LocalDateTime timestamp;
    private BigDecimal filledQuantity;
    private BigDecimal averagePrice;
}
